homeassistant/components/volvooncall/__init__.py

```diff
@@ -215,42 +188,87 @@ def instrument(self, vin, component, attr, slug_attr):
 
     def vehicle_name(self, vehicle):
         """Provide a friendly name for a vehicle."""
-        if (
-            vehicle.registration_number and vehicle.registration_number.lower()
-        ) in self.names:
-            return self.names[vehicle.registration_number.lower()]
-        if vehicle.vin and vehicle.vin.lower() in self.names:
-            return self.names[vehicle.vin.lower()]
-        if vehicle.registration_number:
+        if vehicle.registration_number and vehicle.registration_number != "UNKNOWN":
             return vehicle.registration_number
         if vehicle.vin:
             return vehicle.vin
-        return ""
+        return "Volvo"
+
+    def discover_vehicle(self, vehicle):
+        """Load relevant platforms."""
+        self.vehicles.add(vehicle.vin)
+
+        dashboard = vehicle.dashboard(
+            mutable=self.config[CONF_MUTABLE],
+            scandinavian_miles=self.config[CONF_SCANDINAVIAN_MILES],
+        )
+
+        for instrument in (
+            instrument
+            for instrument in dashboard.instruments
+            if instrument.component in PLATFORMS
+        ):
 
+            self.instruments.add(instrument)
 
-class VolvoEntity(Entity):
+            self.hass.async_create_task(
+                discovery.async_load_platform(
+                    self.hass,
+                    PLATFORMS[instrument.component],
+                    DOMAIN,
+                    (
+                        vehicle.vin,
+                        instrument.component,
+                        instrument.attr,
+                        instrument.slug_attr,
+                        self.coordinator,
```

Comment:
We don't need to pass the coordinator here. Store it in hass.data which we can access in the platforms.